crates/forge/src/test_execution_syscall_handler.rs

```diff
@@ -320,24 +320,36 @@ impl TestExecutionSyscallHandler<'_> {
                 Ok(())
             }
             "start_spoof" => {
-                let contract_address = inputs[0].to_contract_address();
+                let (target, inputs_start) = deserialize_cheat_target(&inputs);
 
-                let version = inputs[1].is_one().then(|| inputs[2].clone());
-                let account_contract_address = inputs[3].is_one().then(|| inputs[4].clone());
-                let max_fee = inputs[5].is_one().then(|| inputs[6].clone());
-                let transaction_hash = inputs[7].is_one().then(|| inputs[8].clone());
-                let chain_id = inputs[9].is_one().then(|| inputs[10].clone());
-                let nonce = inputs[11].is_one().then(|| inputs[12].clone());
+                let version = inputs[inputs_start]
+                    .is_one()
+                    .then(|| inputs[inputs_start + 1].clone());
+                let account_contract_address = inputs[inputs_start + 2]
+                    .is_one()
+                    .then(|| inputs[inputs_start + 3].clone());
+                let max_fee = inputs[inputs_start + 4]
+                    .is_one()
+                    .then(|| inputs[inputs_start + 5].clone());
+                let transaction_hash = inputs[inputs_start + 6]
+                    .is_one()
+                    .then(|| inputs[inputs_start + 7].clone());
+                let chain_id = inputs[inputs_start + 8]
+                    .is_one()
+                    .then(|| inputs[inputs_start + 9].clone());
+                let nonce = inputs[inputs_start + 10]
+                    .is_one()
+                    .then(|| inputs[inputs_start + 11].clone());
 
-                let signature_len = inputs[14]
+                let signature_len = inputs[inputs_start + 13]
                     .to_usize()
                     .expect("Failed to convert signature_len to usize");
```

Comment:
Is it not a problem, we re accessing `inputs_start + 13` not having previously checked if `inputs[inputs_start + 12].is_one()`?

Reply:
in fact it is not

Reply:
What happens if `inputs[inputs_start + 12]` is not `1`? Does that mean that `Option::None` has been passed in `signature` field?

Reply:
exactly

Reply:
Doesn't that further mean that `inputs[inputs_start + 13]` is undefined?

Reply:
it's actually 0

Reply:
https://github.com/foundry-rs/starknet-foundry/issues/1132
i added a task to make it more comprehensible